ANGULAR.md

```diff
@@ -0,0 +1,198 @@
+
+![](./error-6.png)
+
+## Test Code Error
+
+I changed my test to have invalid code:
+
+```ts
+import { GreeterComponent } from './greeter.component'
+
+describe('greeter.component.cy.ts', () => {
+  it('playground', () => {
+    /asd fasdf 
+    cy.mount(GreeterComponent)
+  })
+})
+```
+
+This shows an error in Command Log. This seems better.
+
+![](./error-3.png)
+
+A typing error doesn't show up - I think this is fine, as far as DX goes - since the code does function (TypeScript is running with `transpileOnly`, presumably). The error IS logged to the console.
+
+```ts
+import { GreeterComponent } from './greeter.component'
+
+describe('greeter.component.cy.ts', () => {
+  it('playground', () => {
+    let foo: string = 21
+    cy.mount(GreeterComponent)
+  })
+})
+```
+
+![](./error-4.png)
+
+# Comparison with End to End
+
+It's worth noting E2E is also inconsistent. 
+
+## TypeScript without `@cypress/webpack-preprocessor` - Invalid JS
+
+Error in Command Log.
+
+![](./e2e-2.png)
+
+## TypeScript without `@cypress/webpack-preprocessor` - Type Error
+
+No error (Command Log or console).
+
+## TypeScript with `@cypress/webpack-preprocessor` - Invalid JS
+
+Error in Command Log.
+
+![](./e2e-2.png)
+
+## TypeScript with `@cypress/webpack-preprocessor` - Type Error
+
+Error in overlay.
+
+![](./e2e-1.png)
+
+# Analysis
+
+Broadly speaking, there are two classes of errors:
+
+- App code errors (in the component files, for example). This represents something that would block deployment, since Angular won't compile).
+- Test code errors. This code is not part of the production bundle.
+
+Both categories have two type of coding error:
+  - Invalid code. Example: `let foo const /*** string`
+  - Incorrect types. Example: `const foo: string = 123`
+
+| Code | Type of Error | Current Behavior 
+| -- | -- | -- | 
+| App Code | Invalid JS | console.error | 
+| App Code | Typing Error | console.error |
+| Spec Code | Invalid JS | Command Log | 
+| Spec Code | Typing Error | console.error | 
+
+There is *three* places errors can manifest.
+
+1. Command Log
+2. console
+3. AUT overlay
+
+# Recommendations
+
+## Invalid JS errors (eg, the code cannot be executed):
+
+I'd recommend following End to End Testing's current pattern, which is to show an error in the Command Log, pointing to the location of the error.
+
+I don't think the AUT is a good place for a compilation error. The Command Log (aka the Reporter) should report any problems.
+
+## Typing Errors
+
+I don't think this should show in the Command Log or the AUT. Component Testing, and Cypress in general, is not a static analysis tool. We tell you if you code does what is says it does, regarding rendering and interactions. We should leave typing errors to a the IDE or a separate tsc process. Most build tools have something built in to prevent you from compiling code with typing errors.

```

Comment:
I'd thrown in an argue against this. We should show any compilation errors (TS or JS). The dev shouldn't think they code is able to compile through developing with Cypress only to find out later it doesn't.

Reply:
We would have to turn back on the dev-server overlay (at least in open mode). The drawback is that an error in one spec can affect the other. I really don't see this as being a big deal, a developer will eventually need to fix it sooner or later and sooner is normally better than later